site/content/docs/features/config-file.md

```diff
@@ -1 +1,632 @@
+
+If you use a JSON configuration file in conjunction with the `-s` (`--silent`) option, you can specify deployment options without being prompted by the deployment tool at all.
+
+This section documents the JSON definitions and syntax that you use to construct a configuration file.
+
+## Common JSON parameters
+
+The parameters you can include in a JSON configuration file depend on the type of deployment you're doing, as shown later in this topic. The following parameters are common to all deployment types.
+
+_**AWSProfile**_  
+The name of the AWS profile to use if not using the `[default]` profile.
+
+_**AWSRegion**_  
+The name of the AWS Region to use if not using the region from the `[default]` profile.
+
+_**StackName**_  
+The name of the AWS CloudFormation stack to use for your application. It can be the name of an existing stack or the name of a new stack to create.
+
+_**RecipeId**_  
+A value that identifies the type of deployment you want to perform. See the individual JSON syntax sections below for more information.
+
+## JSON syntax for Amazon ECS deployments
+
+Use this JSON definition to deploy your application to [Amazon Elastic Container Service (Amazon ECS)](https://docs.aws.amazon.com/ecs) with compute power managed by AWS Fargate compute engine. If your project doesn't contain a Dockerfile, it will be automatically generated.
+
+This definition covers the following deployment types, which are provided in the `RecipeId` parameter:
+
+* `AspNetAppEcsFargate`: An ASP.NET Core application. Recommended if you want to deploy your application as a container image on Linux.
+* `ConsoleAppEcsFargateService`: A .NET console application built using a Docker file and deployed as a long running service app. Recommended if you want to deploy a service as a container image on Linux.
+* `ConsoleAppEcsFargateScheduleTask`: A .NET console application built using a Docker file and deployed as a scheduled task. Recommended if you want to deploy a scheduled task as a container image on Linux.
+
+
+```
+{
+    "AWSProfile": "string",
+    "AWSRegion": "string",
+    "StackName": "string",
+    "RecipeId": "AspNetAppEcsFargate" | "ConsoleAppEcsFargateService" | "ConsoleAppEcsFargateScheduleTask",
+    "OptionSettingsConfig":
+    {
+        "ECSCluster":
+        {
+            "CreateNew": boolean,
+            "NewClusterName": "string",
+            "ClusterArn": "string"
+        },
+        "ApplicationIAMRole":
+        {
+            "CreateNew": boolean,
+            "RoleArn": "string"
+        },
+        "Vpc":
+        {
+            "IsDefault": boolean,
+            "CreateNew": boolean,
+            "VpcId": "string"
+        },
+        "TaskCpu": integer,
+        "TaskMemory": integer,
+        "ECSEnvironmentVariables": keyvalue,

```

Comment:
Will the reader understand how to handle a `keyvalue`? I couldn't find it as a normal JSON datatype.

Reply:
I would add an example while we're at it.
```
"ECSEnvironmentVariables": {
   "key1": "value1",
   "key2": "value2"
}
```